### PR TITLE
Don't run composer as root

### PR DESCRIPTION
**Pull request by tiagom62**
Still needs to be thoroughly tested. Please don't merge yet.

### CODE REVIEW

**Comment by snipe** [comment]
Awesome, thank you! I've been wanting to handle this issue for ages and didn't get around to it. Ping me back when it's ready!

**Comment by EarlRamirez** [comment]
@snipe do you think it will be good to add **COMPOSER_PROCESS_TIMEOUT=2000* or any different number for users that have a slow internet connection?

I have the following on the customised OS
# Added composer_process_timeout variable for slow internet connections
 COMPOSER_PROCESS_TIMEOUT=6000 php composer.phar install --no-dev --prefer-source

**Comment by dmeltzer** [comment]
As you're poking at the script, you should be able to use the Ubuntu 16.04 section for 18.04 as well, I made the edit locally and it seems to be behaving.

**Comment by tiagom62** [comment]
@snipe I think i'm done with the changes this round. Tested against the boxes in Vagrantfile.

Composer is installed globally using `snipeitapp` user. The app is now owned by `snipeitapp` user as well.

Let me know what you think and if you want any changes.

File: snipeit.sh [hunk 1]
```diff
@@ -136,58 +131,103 @@ install_packages () {
...
+  if [ "$distro" == "ubuntu" ] || [ "$distro" == "debian" ] ; then
+    adduser --quiet --disabled-password --gecos '""' "$APP_USER"
+  else
+    adduser "$APP_USER"
+  fi
+
+  usermod -a -G "$apache_group" "$APP_USER"
+}
+
+run_as () {
```
Comment (dmeltzer) [inline snipeit.sh]: I might rename this run_as_webuser() or similar.  it's confusing when reading through the code to know that "run_as php -r blah blah blah" does not mean to run the command a user named php.

Comment (tiagom62) [reply snipeit.sh]: Great point, I’ll make the change later tonight.

**Comment by EarlRamirez** [comment]
@tiagom62 saw that it has not be merged with the master; therefore, don't know if you still can update the script and modify line 208 SELinux label _httpd_sys_script_rw_t_, the label should be httpd_sys_rw_content_t at least for the storage and public directory.

**Comment by snipe** [comment]
@tiagom62 I was waiting until it looked like you were done - is this ready for review/merge?

**Comment by tiagom62** [comment]
@EarlRamirez Why is this change needed? That should probably be a separate PR.

@snipe Yup, it is ready for review/merge.

**Pull request merged**